sklearn/metrics/pairwise.py

```diff
@@ -1138,6 +1111,127 @@ def _pairwise_callable(X, Y, metric, **kwds):
                   'sokalsneath', 'sqeuclidean', 'yule', "wminkowski"]
 
 
+def _check_chunk_size(reduced, chunk_size):
+    is_tuple = isinstance(reduced, tuple)
```

Comment:
Adding a comment above that ``reduced`` should be either an array/list/sparse matrix or a tuple of arrays/sequence/.. would help comprehension.

Reply:
Do you think we should have this check, or is it unnecessarily constraining?

Reply:
Well, it's good to have such test, particularly with a user-provided reduce function. Checking each chunk shouldn't be too costly. From my limited understanding of dask, there, they only validate the processing on some sample data (e.g. something the first (few) chunks) then run the full processing without it. But that would add unnecessary complexity here.   Just documenting a bit more this function would help.